syntax = "proto3";

package LAMARCK_ML;

import "LAMARCK_ML/architectures/variables/Variable.proto";
import "LAMARCK_ML/data_util/Attribute.proto";
import "LAMARCK_ML/architectures/IOMapping.proto";

message FunctionProto {
    string class_name = 1;
    string id_name = 2;

    repeated IOMappingProto input_mapping = 5;
    repeated AttributeProto attr = 8;

    repeated VariableProto variables = 10;
}